### DISCUSSION TITLE
Autostart web browser on boot on specific URL in fullscreen mode

### DISCUSSION THREAD

**Discussion by Kehino** [termux/termux-app#4168] [Q&A] [answered]
Hi,
I would like to configure Termux on my old Samsung tablet to open a web browser in [kiosk mode](https://stackoverflow.com/questions/36229684/kiosk-mode-for-chrome-or-firefox) (or similar behaviour) at boot. If I want to rely on Termux, I understand I need the Termux:Boot addon, and maybe the Termux:API addon? (I have found a few similar apps, but I'd like to use Termux as I also need the SHHD backend).

**Comment by williampiti** [comment, marked as answer]
Hi(*゜▽゜)_□

Here’s a setup you can try to launch a web browser in kiosk mode with Termux on boot:

Steps:

1. Install [Termux:Boot](https://f-droid.org/packages/com.termux.boot/): This addon allows scripts in the ~/.termux/boot/ directory to run automatically when the device boots. Install it from the Play Store or F-Droid if you haven’t already.


2. Create a startup script: In Termux, create a new script in the boot directory by running:

```
mkdir -p ~/.termux/boot
nano ~/.termux/boot/start_kiosk.sh
```

3. Script content: In your start_kiosk.sh file, add the following code. It uses the am start command to open Chrome (or your preferred browser) to a specified URL.

#!/data/data/com.termux/files/usr/bin/bash
# Launch Chrome in kiosk mode to open a specific URL

# Delay to ensure the system has booted completely
sleep 10

# Start Chrome in kiosk mode
```
am start --user 0 -n com.android.chrome/com.google.android.apps.chrome.Main -d "http://example.com"

Replace "http://example.com" with the URL you want to display.
```

4. Make the script executable:

`chmod +x ~/.termux/boot/start_kiosk.sh`


5. Testing: Reboot the device to see if Chrome opens automatically at startup. If Chrome doesn’t open in true kiosk mode, you might need to explore other third-party browser apps or use a specialized kiosk app from the Play Store.



Additional Notes:

Using Termux:API: If you need more control over device functions, install Termux:API to issue system commands.

SSHD: Since you mentioned needing SSHD, ensure it’s running by adding sshd to your startup script if necessary.


Hope this helps(T^T)
Lemme know if you run into any issues(pls not(._.))
